src/gt4py/next/program_processors/runners/dace_fieldview/gtir_dataflow.py

```diff
@@ -1309,3 +1320,38 @@ def visit_SymRef(self, node: gtir.SymRef) -> IteratorExpr | MemletExpr | SymbolE
         # if not in the lambda symbol map, this must be a symref to a builtin function
         assert param in gtir_python_codegen.MATH_BUILTINS_MAPPING
         return SymbolExpr(param, dace.string)
+
+    def apply(
+        self,
+        node: gtir.Lambda,
+        args: Sequence[IteratorExpr | MemletExpr | SymbolExpr],
+    ) -> tuple[list[DataflowInputEdge], DataflowOutputEdge]:
+        # lambda arguments are mapped to symbols defined in lambda scope
+        prev_symbols: dict[
+            str,
+            Optional[IteratorExpr | MemletExpr | SymbolExpr],
+        ] = {}
+        for p, arg in zip(node.params, args, strict=True):
+            symbol_name = str(p.id)
+            prev_symbols[symbol_name] = self.symbol_map.get(symbol_name, None)
+            self.symbol_map[symbol_name] = arg
```

Comment:
This needs motivation, why you do this and why you ignore some parts.
I would think it should be simpler to just make a copy of `symbol_map` and then restore it.
I also would have guessed that you could turn it into a dict comprehension, which could be more efficient (maybe).

Reply:
I agree, it is simpler to make a copy of `symbol_map` and then restore it.